python/triton/compiler/code_generator.py

```diff
@@ -1270,7 +1270,7 @@ def kernel_suffix(signature, specialization):
         suffix += str(i)
         if i in specialization.equal_to_1:
             suffix += 'c'
-        if i in specialization.divisible_by_16:
+        if i in specialization.divisibility:
             suffix += 'd'
```

Comment:
I don't understand how it works. Does divisibility means divisible by 16? If it can be other divisibility then would they all have the same suffix?

Reply:
IIUC we only support `divisibility=16` cases before.

Reply:
Hi @ThomasRaoux , I guess it's a matter of design choice. The original implementation was very specific. The specialization for divisibility was setting always the divisibility to 16 (like @antiagainst  is mentioning). 

So now we have  that `AttrsDescriptor.arg_properties` is a `map(attr_name, param)` and  all the params will be set to a single value in `property_values` (so that each element in `param_set` will have that single value)

We can decide to be more generic and have a `arg_properties` like a `map(attr_name, (param_set, value))` and get rid of the `property_values` map.

Reply:
my point was more: either this is specific to divisibility16 and it should be explicit in the naming or the suffix should be different based on the divisibility value?

Reply:
Alright, if it is how we represent the attribute from the class, I made the change that now the slots are `divisibility_16` and `equal_to_1`, so it should be clearer what we are asking for